frontend/src/flux/actions/authActions.js

```diff
@@ -0,0 +1,110 @@
+import axios from 'axios';
+import {
+  AUTH_ERROR,
+  LOGIN_FAIL,
+  LOGIN_SUCCESS,
+  LOGOUT_SUCCESS,
+  REGISTER_FAIL,
+  REGISTER_SUCCESS,
+  USER_LOADED,
+  USER_LOADING,
+} from './types';
+import { returnErrors } from './errorActions';
+
+export const loadUser = () => (dispatch, getState) => {
+  dispatch({ type: USER_LOADING });
+
+  axios
+    .get('/api/users', tokenConfig(getState))
+    .then(res =>
+      dispatch({
+        type: USER_LOADED,
+        payload: res.data,
+      })
+    )
+    .catch(err => {
+      dispatch(returnErrors(err.response.data, err.response.status));
+      dispatch({
+        type: AUTH_ERROR,
+      });
+    });
+};
+
+export const register = ({ name, email, password }) => dispatch => {
+  const config = {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  };
+
+  const body = JSON.stringify({ name, email, password });
+
+  axios
+    .post('/api/users', body, config)
+    .then(res =>
+      dispatch({
+        type: REGISTER_SUCCESS,
+        payload: res.data,
+      })
+    )
+    .catch(err => {
+      dispatch(
+        returnErrors(err.response.data, err.response.status, 'REGISTER_FAIL')
+      );
+      dispatch({
+        type: REGISTER_FAIL,
+      });
+    });
+};
+
+export const login = ({ name, password }) => dispatch => {
+  const config = {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  };
+
+  const body = JSON.stringify({ name, password });
+
+  axios
+    .post('http://localhost:3000/api/session', body, config)
```

Comment:
@marcianszlovak the backend protocol, url and port should be in a constant `BACKEND_URL`